### PR TITLE
Closes AgileVentures/MetPlus_tracker#484

### PR DESCRIPTION
**Pull request by jeanwei**
Second PR for the same feature branch

### CODE REVIEW

File: app/views/job_seekers/home.html.haml [hunk 1]
```diff
@@ -21,7 +21,6 @@
   Job Opportunities - New
   %span.text-muted.small (Jobs since last login shown in yellow)
 %div
-  .pagination-div{ id: 'recent-jobs',
-                   data: { url: list_jobs_path(@recent_jobs_type,
-                           js_login: @js_last_sign_in ) } }
```
Comment (joaopapereira) [inline app/views/job_seekers/home.html.haml]: We no longer need the last sign in date?

File: app/controllers/jobs_controller.rb [hunk 2]
```diff
@@ -105,21 +102,6 @@ def destroy
     redirect_to jobs_url
   end
 
-  def applications
-    @application_type = params[:application_type] || 'job-applied'
-  end
-
-  def applications_list
```
Comment (joaopapereira) [inline app/controllers/jobs_controller.rb]: We no longer use this functionalities?

File: app/views/jobs/_list_all.html.haml [hunk 3]
```diff
@@ -37,8 +37,6 @@
                         company_person_path(job.company_person))
           - when :num_applicants
             %td.col-md-1
-              = link_to_unless(job.number_applicants == 0,
-                               job.number_applicants,
-                               applications_job_path(job.id))
```
Comment (joaopapereira) [inline app/views/jobs/_list_all.html.haml]: Is this link somewhere else?

File: app/views/jobs/show.html.haml [hunk 4]
```diff
@@ -37,9 +42,17 @@
...
-        =link_to "Click Here To Apply Online", '#', class: "col-sm-offset-4",
+      - else
+        = link_to "Click Here To Apply Online", '#', class: "col-sm-offset-4",
                   data: { :toggle => 'modal',
-                          :target => '#applyJobModal'}
-        =render partial: 'apply_modal', :locals => {:job => @job}
+                          :target => '#applyJobModal' }
+        = render partial: 'apply_modal', :locals => { :job => @job }
+
+  - if pets_user.is_a?(CompanyPerson) && (pets_user.company == @job.company)
```
Comment (joaopapereira) [inline app/views/jobs/show.html.haml]: Should we allow agency people to see applicants?

File: app/controllers/jobs_controller.rb (hunk 2 above)
Comment (jeanwei) [inline app/controllers/jobs_controller.rb]: #applications ('/jobs/:id/applications') has been moved to #show ('jobs/:id') as #applications is to list job applications for a given job.
So, the #applications view is integrated to  #show view with conditions.

File: app/views/jobs/show.html.haml (hunk 4 above)
Comment (jeanwei) [inline app/views/jobs/show.html.haml]: ok. I will take notes of this in my authorizationg story

File: app/views/job_seekers/home.html.haml (hunk 1 above)
Comment (jeanwei) [inline app/views/job_seekers/home.html.haml]: we are using the last sign in date. Just write in another way to keep the argument simple
Please refer to  app/views/jobs/_compact_list.haml, line 18

File: app/views/jobs/_list_all.html.haml (hunk 3 above)
Comment (jeanwei) [inline app/views/jobs/_list_all.html.haml]: Previous title link:
previous_title

Previous applicants link:
screen shot 2016-11-18 at 14 38 14

Current title link:
screen shot 2016-11-18 at 14 35 09
which is the combination of previous title & previous applicants link

**Comment by jeanwei** [comment]
Btw. this PR is the same PR which was being asked to rebase before able to merge. patrick and I re-create this new one because the houndci check seemed to stuck and reported 200 errors after rebase.

**Comment by patmbolger** [comment]
I had reviewed these changes in the prior PR and my requested changes at that time were all accommodated.  The only remaining changes that I would like to see is to address the houndci issues (other than "line too long").

**Pull request merged**